pkg/util/nodeimages/nodeimages_util.go

```diff
@@ -79,10 +81,28 @@ func GetNodeImagesForJob(reader client.Reader, job *appsv1alpha1.ImagePullJob) (
 			return nil, err
 		}
 
-		nodeImageNames := sets.NewString()
+		pods := make([]*v1.Pod, 0, len(podList.Items))
 		for i := range podList.Items {
 			pod := &podList.Items[i]
-			if !kubecontroller.IsPodActive(pod) || pod.Spec.NodeName == "" || nodeImageNames.Has(pod.Spec.NodeName) {
+			if !kubecontroller.IsPodActive(pod) || pod.Spec.NodeName == "" {
```

Comment:
why drop the nodeImage check? sorting with less pods  can be more efficient

Reply:
we can not distinct pods with same node here, it may cause incorrect pre-download sequence